src/__test__/pages/experiments/[experimentId]/data-processing/data-integration/CalculationConfig.test.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import {
+  Spin, Form,
+} from 'antd';
+import { mount, configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import preloadAll from 'jest-next-dynamic';
+import { Provider } from 'react-redux';
+
+import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
+
+import CalculationConfig from '../../../../../../pages/experiments/[experimentId]/data-processing/data-integration/components/CalculationConfig';
+import initialExperimentState, { initialProcessingState } from '../../../../../../redux/reducers/experimentSettings/initialState';
+
+jest.mock('localforage');
+enableFetchMocks();
+const mockStore = configureStore([thunk]);
+
+describe('Data Integration Calculation Config', () => {
+  const storeState = {
+    experimentSettings: {
+      ...initialExperimentState,
+      processing: initialProcessingState,
+    },
+  };
+
+  configure({ adapter: new Adapter() });
+
+  beforeEach(async () => {
+    await preloadAll();
+
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: jest.fn().mockImplementation((query) => ({
+        matches: false,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(), // deprecated
+        removeListener: jest.fn(), // deprecated
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
+
+    const response = new Response(
+      JSON.stringify(
+        {
+          processingConfig:
+            { ...initialProcessingState },
+        },
+      ),
+    );
+
+    fetchMock.resetMocks();
+    fetchMock.doMock();
+    fetchMock.mockResolvedValue(response);
+  });
+
+  it('renders correctly when nothing is loaded', () => {
+    const store = mockStore({
+      embeddings: {},
+      experimentSettings: {
+        ...initialExperimentState,
+      },
+    });
+
+    const component = mount(
+      <Provider store={store}>
+        <CalculationConfig
+          experimentId='1234'
+          width={50}
+          height={50}
+        />
+      </Provider>,
+    );
+
+    // As there is no state implemented yet
+    // There should be a form loaded.
+    const form = component.find(Form);
+    expect(form.length).toBeGreaterThan(0);
+  });
+
+  it('renders correctly when the data is in the store', () => {
+    const store = mockStore(storeState);
+
+    const component = mount(
+      <Provider store={store}>
+        <CalculationConfig
+          experimentId='1234'
```

Comment:
Why are experimentId, width and length set as props? There is probably a good reason that I don't know about because I have seen them in many other tests too.

Reply:
I believe most of them are just copy pasted from other test codes 😆 They aren't read by the components because they aren't used. I'll take them out.

Reply:
The experiment id is needed for all components that dispatch redux actions. The reason why they are needed is because when dispatching a redux action, the ui usually sends a request to the backend, and for that request we need to have the experiment id.
In data-epxloration components we have to have width and height as props because of the tiled component (called react mosaic) that we use to render the data exploration page into tiles. If you look at the code in here: https://github.com/biomage-ltd/ui/blob/master/src/pages/experiments/%5BexperimentId%5D/data-exploration/index.jsx#L54 the mosaic framework passes width and height to each of the components in the `TILE_MAP` object, which are calculated from the mosaic framework based on the screen width and height and the percentage split of how much proportion of the screen each component should take. 
If you are more interested, you can take a look in here too: https://github.com/biomage-ltd/ui/blob/master/src/redux/actions/layout.js

Reply:
so in short, width and height are not needed for components outside of data exploration. Experiment id is needed for every component that has to send a request to the backend at some point.

Reply:
Deleted in new commit